addons/web/static/src/core/debug/profiling/profiling_qweb.js

```diff
@@ -0,0 +1,325 @@
+                // Open tag.
+                const nodeTagName = node.nextElementSibling;
+                const aceLine = nodeTagName.parentNode;
+                const index = [].indexOf.call(aceLine.parentNode.children, aceLine);
+                const row = rows[index];
+
+                // Add a children to the arch and compute the xpath.
+                xpath += "/" + nodeTagName.textContent;
+                let i = 1;
+                while (flat[xpath + "[" + i + "]"]) {
+                    i++;
+                }
+                xpath += "[" + i + "]";
+                flat[xpath] = { xpath: xpath, tag: nodeTagName.textContent, children: [], directive: [] };
+                arch.push(flat[xpath]);
+                parent.children.push(flat[xpath]);
+
+                // Compute delay and query number.
+                const closed = !!row.querySelector(".ace_closed");
+                const delays = [];
+                const querys = [];
+                const groups = {};
+                let displayDetail = false;
+                for (const line of this.profileLines) {
+                    if (
+                        line.view_id === this.state.viewID &&
+                        (closed ? line.xpath.startsWith(xpath) : line.xpath === xpath)
+                    ) {
+                        delays.push(line.delay);
+                        querys.push(line.query);
+                        const directive = line.directive.split("=")[0];
+                        if (!groups[directive]) {
+                            groups[directive] = {
+                                delays: [],
+                                querys: [],
+                            };
+                        } else {
+                            displayDetail = true;
+                        }
+                        groups[directive].delays.push(this._formatDelay(line.delay));
+                        groups[directive].querys.push(line.query);
+                    }
+                }
+
+                // Display delay and query number in front of the line.
+                if (delays.length && !row.querySelector(".o_info")) {
+                    this._renderInfo(delays, querys, displayDetail, groups, row);
+                }
+            }
+            node.setAttribute("data-xpath", xpath);
+        });
+    }
+    /**
+     * Set the view ID and send atch to ACE.
+     *
+     * @private
+     */
+    _renderView() {
+        const view = this.viewObjects.find((view) => view.id === this.state.viewID);
+        if (view) {
+            const arch = this.viewArch[view.id] || "";
+            if (this.aceSession.getValue() !== arch) {
+                this.aceSession.setValue(arch);
+            }
+            const content = this.selector.el.querySelector(".dropdown-menu a[data-id='" + view.id + "']").innerHTML;
+            this.selector.el.querySelector(".dropdown-toggle").innerHTML = content;
```

Comment:
Not looking great, would be better to see if there is a way around this by using the template and less DOM manips